pkg/openstack/instances.go

```diff
@@ -625,5 +664,64 @@ func getAttachedInterfacesByID(client *gophercloud.ServiceClient, serviceID stri
 		return interfaces, err
 	}
 
+	// Check if trunk ports are attached
+	var subports []trunks.Subport
+
+	listOpts := trunks.ListOpts{}
+	allPages, err := trunks.List(network, listOpts).AllPages()
+	if err != nil {
+		klog.Errorf("Failed to list trunks: %v", err)
+		return interfaces, nil
+	}
+	allTrunks, err := trunks.ExtractTrunks(allPages)
+	if err != nil {
+		klog.Errorf("Failed to extract trunks: %v", err)
+		return interfaces, nil
+	}
+
+	for _, trunk := range allTrunks {
+		for _, iface := range interfaces {
+			if iface.PortID == trunk.PortID {
+				// Get subports attached to the trunk
+				s, err := trunks.GetSubports(network, trunk.ID).Extract()
+				if err != nil {
+					klog.Errorf("Failed to get subports for trunk %s: %v", trunk.ID, err)
+					return interfaces, nil
+				}
+				subports = append(subports, s...)
+			}
+		}
+	}
+
+	klog.V(5).Infof("subports %v", subports)
+
+	for _, sport := range subports {
+		p, err := ports.Get(network, sport.PortID).Extract()
+		if err != nil {
+			return interfaces, nil
+		}
+		n, err := networks.Get(network, p.NetworkID).Extract()
+		if err != nil {
+			return interfaces, nil
+		}
+		var iface = attachinterfaces.Interface{
+			PortState: "ACTIVE",
+			FixedIPs:  []attachinterfaces.FixedIP{},
+			PortID:    p.ID,
+			NetID:     n.Name,
```

Comment:
Why it's setting network name as NetID?

Reply:
This is because [Networking] config takes network name not network id. IPs on subports need be associated with network names to allow the filtering.
Do you see a better way?